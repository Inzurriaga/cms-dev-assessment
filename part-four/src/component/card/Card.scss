.card {
  position: relative;

  div {
    position: absolute;
    top: 0;
    left: 0;
    padding: 5px;
    background-color: #ffffff;
    display: flex;

    p {
      margin: 0;
      font-size: 0.8em;
    }

    h2 {
      margin: 0 5px;
      font-size: 1em;
      font-weight: 900;
    }
  }
  
  img {
    width: 100%;
    height: 100%;
  }
}

.card-exit {
  animation: exitAnim 0.5s linear forwards;
}

.card-enter {
  animation: enterAnim 0.5s linear forwards;
}

.page-0-grid-0, .page-2-grid-0 {
  grid-column: 1 / span 2 ;
  grid-row: 1 / span 2 ;
}

.page-0-grid-1 , .page-2-grid-1{
  grid-column: 1 / span 2 ;
  grid-row: 3 / span 2 ;  
}

.page-0-grid-2 {
  grid-column: 3 / span 2 ;
  grid: span 1;
}

.page-0-grid-3 {
  grid-column: 3 / span 2 ;
  grid-row: 2 / span 2;
}

.page-0-grid-4 {
  grid-column: 3 / span 2 ;
  grid: span 1;
}

.page-1-grid-0 {
  grid-column: 1;
  grid-row: 1 / span 2;
}

.page-1-grid-1 {
  grid-column: 1;
  grid-row: 3 / span 2;
}

.page-1-grid-2 {
  grid-column: 2 / span 2;
  grid-row: 1 / span 2;
}

.page-1-grid-3 {
  grid-column: 2 / span 2;
  grid-row: 3 / span 2;
}

.page-1-grid-4 {
  grid-column: 4;
  grid-row: 1 / span 4;
}

.page-2-grid-2, .page-3-grid-1 {
  grid-column: 3 / span 2;
  grid-row: 1 / span 4;
}

.page-3-grid-0 {
  grid-column: 1 / span 2;
  grid-row: 1 / span 4;
}

@keyframes exitAnim {
  0% {
    clip-path: polygon(0 0, 100% 0%, 100% 100%, 0% 100%);
  }
  50% {
    clip-path: polygon(0 55%, 100% 45%, 100% 100%, 0 100%);
  }
  100% {
    clip-path: polygon(0 100%, 100% 100%, 100% 100%, 0% 100%);
  }
}

@keyframes enterAnim {
  0% {
    clip-path: polygon(0 100%, 100% 100%, 100% 100%, 0% 100%);
  }
  50% {
    clip-path: polygon(0 55%, 100% 45%, 100% 100%, 0 100%);
  }
  100% {
    clip-path: polygon(0 0, 100% 0%, 100% 100%, 0% 100%);
  }
}
