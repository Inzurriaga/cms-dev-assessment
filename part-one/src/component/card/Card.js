import React, { Component } from 'react';
import fallBack from '../../fallback.jpg';

export default class Card extends Component {
  
  constructor() {
    super();
    this.state = {
      descriptionView: 'Read More'
    };
  }

  toggleDescriptionView = () => {
    if(this.state.descriptionView === 'Read More') {
      this.setState({
        descriptionView: 'Read Less'
      })
    }else {
      this.setState({
        descriptionView: 'Read More'
      })
    }
  }

  fallBackImage = (e) => {
    e.target.src = fallBack;
  }

  render() {
    const { description, mediaurl, title } = this.props.data;
    const { descriptionView } = this.state;
    const cssStyle = descriptionView.split(" ")[1]
    return (
      <article className='card'>
        <img src={ mediaurl } onError={ this.fallBackImage } alt={ title }/>
        <div className={cssStyle}>
          <h2>{ title }</h2>
          <p>{ description }</p>
          <button onClick={ this.toggleDescriptionView }>{ descriptionView }</button>
        </div>
      </article>
    );
  }
}