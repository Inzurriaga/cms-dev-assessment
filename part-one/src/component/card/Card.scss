.card {
  height: 100%;
  width: 100%;
  position: relative;
  overflow: hidden;

  img {
    width: 100%;
    height: 319px;
  }

  div {
    overflow: hidden;
    box-sizing: border-box;
    position: relative;

    h2 {
      font-weight: 900;
    }
  }

  .Less {
    min-height: 319px;
    padding-bottom: 100px;
  }

  .More {
    height: 319px;

    &:after {
      content: '';
      position: absolute;
      z-index: 1;
      bottom: 0;
      left: 0;
      background-image: linear-gradient(to bottom, 
                        rgba(255,255,255, 0), 
                        rgba(255,255,255, 0.50), 
                        rgba(255,255,255, 1) 50%);
      width: 100%;
      height: 10em;
    }
  }

  button {
    cursor: pointer;
    transition: 0.25s;
    font-family: 'Lato', sans-serif;
    font-weight: 400;
    font-size: 1.2em;
    color: #FFFFFF;
    border: none;
    outline: none;
    padding: 10px;
    border-radius: 5px;
    position: absolute;
    transform: translate( -50%, 100%);
    z-index: 3;
    bottom: 0;
    left: 50%;
    background-color: #FF0000;

    &:hover {
      background-color: #BA0000;
    }
  }

  &:hover {

    button {
      transform: translate( -50%, -100%);
    }
  }
}

@media screen and (min-width: 1026px ) {

  .card {

    &:nth-child(6n + 1) {
      grid-column-end: span 2;
    }

    &:nth-child(6n) {
      display: grid;
      grid-column-end: span 2;
      grid-template-columns: repeat(2, 50%);
      grid-template-rows: 1fr;

      div {
        padding: 0 10px 0 10px;
        height: 100%;
      }
      
      img {
        height: 638px;
      }
    }
  }
}