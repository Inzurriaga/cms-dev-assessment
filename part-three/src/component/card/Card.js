import React, { Component } from 'react';
import fallBack from '../../fallback.jpg';

export default class Card extends Component {

  fallBackImage = (e) => {
    e.target.src = fallBack;
  }

  render() {
    const { title, number, mediaurl } = this.props.data;
    const { gridPlacement, animation} = this.props;
    
    return (
      <article className={ `card ${gridPlacement} card-${animation}` }>
        <div>
          <p>{ `0${number}.` }</p>
          <h2>{ title }</h2>
        </div>
        <img src={ mediaurl } onError={ this.fallBackImage } alt={ title } />
      </article>
    );
  }
}