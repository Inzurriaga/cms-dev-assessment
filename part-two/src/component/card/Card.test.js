import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

const mockData = {
  title: 'test',
  description: 'hello world',
  mediaurl: 'http://www.hello.com'
};

describe('Card', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Card data={ mockData }/>
    );
  });

  it('should have a snap shot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});