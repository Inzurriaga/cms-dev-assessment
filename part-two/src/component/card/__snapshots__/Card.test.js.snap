// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should have a snap shot 1`] = `ShallowWrapper {}`;
